modules/space/shaders/star_ge.glsl

```diff
@@ -24,85 +24,230 @@
+        scaleMultiply                  = ((lumCent * (ge_bvLumAbsMagAppMag.y + 5E9)) + 
+                                         (radiusCent * double(starRadius)) + 
+                                         (brightnessCent * apparentBrightness * 5E15)) * 
+                                         pow(10.0, magnitudeExponent);
+    } else if (psfParamConf == 3) {
+        float absMag = ge_bvLumAbsMagAppMag.z;
+        scaleMultiply = (-absMag + 35.f) * pow(10.0, magnitudeExponent + 8.5f);
+    } else if (psfParamConf == 4) {
+        float absMag = vs_bvLumAbsMagAppMag[0].z;
+        double distanceToStarInMeters = length(dpos.xyz - eyePosition);
+        double distanceToCenterInMeters = length(eyePosition);
+        float distanceToStarInParsecs = float(distanceToStarInMeters/PARSEC); 
+        //float appMag = absMag + 5*log(distanceToStarInParsecs) - 5.0;
+        float appMag = absMag + 5.0 * (log(distanceToStarInParsecs/10.0)/log(2.0));
+        //appMag = vs_bvLumAbsMagAppMag[0].w;
+        
+        //scaleMultiply = (30.623 - appMag) * pow(10.0, magnitudeExponent + 7.0);// * 
+        //float(distanceToStarInMeters/distanceToCenterInMeters);
+        
+        //scaleMultiply = (-appMag + 50.f) * pow(10.0, magnitudeExponent + 7.5f);
+        //scaleMultiply =  log(35.f + appMag) *  pow(10.0, magnitudeExponent + 6.5f);
+        //scaleMultiply = exp((35.f - appMag) * 0.2) * pow(10.0, magnitudeExponent + 2.5f);
+        //scaleMultiply = appMag * pow(10.0, magnitudeExponent + 8.5f);
+        scaleMultiply = exp((-30.0 - appMag) * 0.45) * pow(10.0, magnitudeExponent + 8.f);
+        //scaleMultiply = pow(10.0, (appMag - absMag)*(1.0/5.0) + 1.0) * pow(10.0, magnitudeExponent + 3.f);
+    } else if (psfParamConf == 5) {
+        float absMag = ge_bvLumAbsMagAppMag.z;
+        scaleMultiply = exp((-30.623 - absMag) * 0.462) * pow(10.0, magnitudeExponent + 12.5f) * 2000;
     }
 
-    EndPrimitive();
+    dvec3 scaledRight    = dvec3(0.0);
+    dvec3 scaledUp       = dvec3(0.0);
+    vec4 bottomLeftVertex, bottomRightVertex, topLeftVertex, topRightVertex;
+  
+    // if (distanceToStarInParsecs > 1800.0) {
+    //     scaledRight    = scaleMultiply * invariantRight * 0.5f;
+    //     scaledUp       = scaleMultiply * invariantUp * 0.5f;
+    // } else {
+        dvec3 normal   = normalize(eyePosition - dpos.xyz);
+        dvec3 newRight = normalize(cross(cameraUp, normal));
+        dvec3 newUp    = cross(normal, newRight);
+        scaledRight    = scaleMultiply * newRight;
+        scaledUp       = scaleMultiply * newUp;
+    //}
+   
+    bottomLeftVertex = z_normalization(vec4(cameraViewProjectionMatrix * 
+                        dvec4(dpos.xyz - scaledRight - scaledUp, dpos.w)));
+    gs_screenSpaceDepth  = bottomLeftVertex.w;
+    
+    topRightVertex = z_normalization(vec4(cameraViewProjectionMatrix * 
+                            dvec4(dpos.xyz + scaledUp + scaledRight, dpos.w)));        
+
+    // Testing size:
```

Comment:
Should this still be here? If so, can you add some more comments on how to use it

Reply:
Although it is not necessary there, I prefer to wait to remove all the instances os the z_normalization at once. It can yield small changes here and there.

Reply:
Oh, I didn’t mean the z normalization: I meant all the code that is commented out

Reply:
Oh, I didn’t mean the z normalization: I meant all the code that is commented out

Reply:
Ow... The code can be removed. I didn't remove it yet because it can be used for debugging procedures and I am still waiting for a round of validation with Carter and Brian.